src/utils/time.ts

```diff
@@ -37,14 +39,24 @@ function compareTime(a: number[], b: number[]) {
     return 1;
 }
 
-export function isInTimeInterval(date: Date, time0: string, time1: string) {
+function dummyTime() {
```

Comment:
I will remove this `dummyTime` function in future commits once `isInTimeInterval` and `isNightAtLocation` can calculate exact time.

Reply:
For the first portion of converting: 
[patches.tar.gz](https://github.com/darkreader/darkreader/files/6874024/patches.tar.gz)

The always night/day time requires some more specialized Math. Otherwise we could just fallback to check it every day.

Reply:
The calculation for it would be more or less brute-forcing which is something I don't like so you could just fallback to check it every day, even though if someone was in this time-range they happen for around ~180 days before going back to the correct cycle. Which is another way to calculate the approximate.

Reply:
I excluded the actual calculation changes from this PR to simplify reviewing this PR, however I can add them in if you prefer.

> The calculation for it would be more or less brute-forcing which is something I don't like

As far as I see, calculating sunset/sunrise would be a brute-force for each _day_ of seasonal day/night, which is still better than brute-forcing every 10 seconds. So falling back to checking every day might not be that bad. However, I'd prefer to just fall back to scheduling the next alarm in 24 hours and defering this problem until later.

Reply:
> I excluded the actual calculation changes from this PR to simplify reviewing this PR, however I can add them in if you prefer.

Just add them now. As in both cases I will just approve them and merge them.

>  I'd prefer to just fall back to scheduling the next alarm in 24 hours and defering this problem until later.

👍🏽

Reply:
I just added time calculation for isInTimeInterval (with tests). Solar dawn/sunset is next.

Reply:
I added support for calculating solar dawn/sunset and tests. The results make sense (they fall within reasonable ballpark), but do not match exactly any calculators I could find on the web. I suspect that these slight variations (within an hour) can be caused by different definitions of "dawn" and "sunset" in different models. E.g., someone might say dawn is when you see first sunrays creating ambient light, someone might say it is when sun appears on the horizon and someone might say it is when sun is above horizon and provides reasonable illumination.